metricflow-semantics/metricflow_semantics/query/validation_rules/metric_time_requirements.py

```diff
@@ -18,18 +23,33 @@
 from metricflow_semantics.query.issues.parsing.offset_metric_requires_metric_time import (
     OffsetMetricRequiresMetricTimeIssue,
 )
-from metricflow_semantics.query.resolver_inputs.query_resolver_inputs import ResolverInputForQuery
+from metricflow_semantics.query.issues.parsing.scd_requires_metric_time import (
+    SCDRequiresMetricTimeIssue,
+)
+from metricflow_semantics.query.resolver_inputs.query_resolver_inputs import (
+    ResolverInputForQuery,
+)
 from metricflow_semantics.query.validation_rules.base_validation_rule import PostResolutionQueryValidationRule
 from metricflow_semantics.specs.time_dimension_spec import TimeDimensionSpec
 
 
+@dataclass(frozen=True)
+class QueryItemsAnalysis:
+    """Contains data about which items a query contains."""
+
+    scds: Sequence[str]
```

Comment:
Also - can we store the `LinkableInstanceSpecs` (`TimeDimensionSpec`/`DimensionSpec`/`EntitySpec`)  instead of the qualified names here? It seems fragile to just pass around the string names, especially since it's not always clear that they are qualified names.
Then you can pass the specs into `SCDRequiresMetricTimeIssue` and parse out the qualified names within that class.

Reply:
Will do!